aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,198 @@
+        throw new NoSuchTableException("Cannot find Glue table %s after refresh, " +
+            "maybe another process deleted it or revoked your access permission", tableName());
+      }
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    boolean exceptionThrown = true;
+    boolean isUpdate = false;
+    Table glueTable = null;
+    try {
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        isUpdate = true;
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Cannot commit because table %s already exists in Glue", tableName());
+        }
+      }
+
+      checkMetadataLocation(isUpdate, glueTable, base);
+      Map<String, String> properties = prepareProperties(isUpdate, glueTable, newMetadataLocation);
+      persistGlueTable(isUpdate, properties);
+      exceptionThrown = false;
+    } catch (CommitFailedException | AlreadyExistsException e) {
+      throw e;
+    } catch (ConcurrentModificationException e) {
+      throw new CommitFailedException(e,
+          "Cannot commit %s because Glue detected concurrent update", tableName());
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      throw new AlreadyExistsException(e,
+          "Cannot commit %s because its Glue table already exists when trying to create one", tableName());
+    } catch (SdkException e) {
+      throw new CommitFailedException(e, "Cannot commit %s because unexpected exception contacting AWS", tableName());
+    } finally {
+      if (exceptionThrown) {
+        io().deleteFile(newMetadataLocation);
+      }
+    }
+  }
+
+  private void checkMetadataLocation(boolean isUpdate, Table glueTable, TableMetadata base) {
+    String glueMetadataLocation = isUpdate ? glueTable.parameters().get(METADATA_LOCATION_PROP) : null;
+    String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+    if (!Objects.equals(baseMetadataLocation, glueMetadataLocation)) {
+      throw new CommitFailedException(
+          "Cannot commit %s because base metadata location '%s' is not same as the current Glue location '%s'",
+          tableName(), baseMetadataLocation, glueMetadataLocation);
+    }
+  }
+
+  private Optional<Table> getGlueTable() {
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(databaseName)
+          .name(tableName)
+          .build());
+      return Optional.ofNullable(response.table());
```

Comment:
Why does this use `ofNullable`? Seems like the response returning a null table is an error.